src/V3SignalStrength.cpp

```diff
@@ -0,0 +1,145 @@
+                                                    new AstConst(fl, AstConst::WidthedValue(), 1,
+                                                                 compareConstp))),
+                        new AstConst(fl, strengthLevel),
+                        new AstVarRef(fl, strengthVarp, VAccess::READ)));
+    }
+
+    virtual void visit(AstNodeModule* nodep) override {
+        UINFO(8, nodep << endl);
+        iterateChildren(nodep);
+        for (auto& varpAssigns : m_assigns) {
+            AstVar* varp = varpAssigns.first;
+            FileLine* varFilelinep = varp->fileline();
+            Assigns assigns = varpAssigns.second;
+            if (assigns.size() > 1) {
+                AstVar* strength0Varp = new AstVar(
+                    varFilelinep, VVarType::MODULETEMP, varp->name() + "__s0", VFlagChildDType(),
+                    new AstBasicDType(varFilelinep,
+                                      VBasicDTypeKwd::INTEGER));  // * varp->width());
+                AstVar* strength1Varp = new AstVar(
+                    varFilelinep, VVarType::MODULETEMP, varp->name() + "__s1", VFlagChildDType(),
+                    new AstBasicDType(varFilelinep,
+                                      VBasicDTypeKwd::INTEGER));  // * varp->width());
+                nodep->addStmtp(strength0Varp);
+                nodep->addStmtp(strength1Varp);
+                AstBegin* strengthBlockp
+                    = new AstBegin(varFilelinep, "strength_computing_block", nullptr);
+                for (size_t i = 0; i < assigns.size(); i++) {
+                    int strength0Level, strength1Level;
+                    if (AstStrengthSpec* strengthSpec
+                        = VN_CAST(assigns[i]->strengthSpecp(), StrengthSpec)) {
+                        strength0Level = strengthSpec->strength0p()->strengthLevel;
+                        strength1Level = strengthSpec->strength1p()->strengthLevel;
+                    } else {
+                        strength0Level = 6;  // default strength in strong (6)
+                        strength1Level = 6;
+                    }
+
+                    FileLine* filelinep = assigns[i]->fileline();
+                    if (strength0Level != 0)
+                        strengthBlockp->addStmtsp(
+                            getStrengthAssignmentp(filelinep, strength0Varp, strength0Level,
+                                                   assigns[i]->rhsp()->cloneTree(false), 0));
+                    if (strength1Level != 0)
+                        strengthBlockp->addStmtsp(
+                            getStrengthAssignmentp(filelinep, strength1Varp, strength1Level,
+                                                   assigns[i]->rhsp()->cloneTree(false), 1));
+
+                    assigns[i]->unlinkFrBack();
+                }
+                nodep->addStmtp(new AstInitial(varFilelinep, strengthBlockp));
+
+                AstVarRef* varRefp = new AstVarRef(varFilelinep, varp, VAccess::WRITE);
```

Comment:
So do we only handle 1-bit signals? Or am I missing something?

Reply:
Yes, for now only 1-bit signals are supported. I am working on multidimensional now.